### ISSUE TITLE
PageData typing breaks sveltekit satisfies PageServerLoad

### ISSUE THREAD

**Issue by leon** [nextauthjs/next-auth#9168]
### Environment

  System:
    OS: macOS 14.1
    CPU: (12) arm64 Apple M2 Max
    Memory: 480.56 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.1.0 - ~/.nvm/versions/node/v21.1.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v21.1.0/bin/yarn
    npm: 10.2.0 - ~/.nvm/versions/node/v21.1.0/bin/npm
    pnpm: 8.10.5 - ~/.nvm/versions/node/v21.1.0/bin/pnpm
    bun: 1.0.1 - ~/.bun/bin/bun
  Browsers:
    Chrome: 119.0.6045.159
    Safari: 17.1
    Safari Technology Preview: 17.4

  npmPackages:
    @auth/core: ^0.18.1 => 0.18.1 
    @auth/sveltekit: ^0.3.12 => 0.3.12

### Reproduction URL

https://github.com/leon/bug-sveltekit-authjs

### Describe the issue

auth.js adds this as typings for PageData
this causes typescript to error out

```ts
export declare function SvelteKitAuth(
	options: SvelteKitAuthConfig | DynamicSvelteKitAuthConfig
): Handle;
declare global {
	namespace App {
		interface Locals {
			getSession(): Promise<Session | null>;
		}
		interface PageData {
			session: Session | null;
		}
	}
}
```

when writing load functions like this

```ts
export const load = (async () => {
	return {};
}) satisfies PageServerLoad;
```

we need to make session optional in the interface

### How to reproduce

open the repo and go to the file +page.server.ts
then you should see the red squiggly line under `satisfies`

### Expected behavior

no squiggly 🤓

**Comment by eagerestwolf** [comment]
Interesting. I can confirm the bug using the repro repository, but this bug does not exist on my project. The difference? I am using a `+layout.server.ts` to setup the `page` store.  It seems that somewhere in the server load chain, the `session` **must** be populated. For example, this code works fine on my end (ignore the `loadFlash` function, my app uses `sveltekit-flash-message`, and I have my own issues with that breaking because it has type issues with `LayoutServerLoad`):

`src/routes/+layout.server.ts`:
```ts
import type { Session } from "@auth/core/types";
import { loadFlash } from "sveltekit-flash-message/server";

import type { LayoutServerLoad } from "./$types";

export const load: LayoutServerLoad = loadFlash(async ({ locals }) => {
  const getSession = async (): Promise<Session | null> => {
    return locals.getSession();
  };

  return { session: getSession() };
});
```

`src/routes/test/path/+page.server.ts`:
```ts
import type { PageServerLoad } from "./$types";

export const load = (async () => {
  return {};
}) satisfies PageServerLoad; // <== no error, since I defined `session` in my +layout.server.ts
```

If my theory is correct, perhaps the issue could be resolved by doing one of three things. First, the `session` attribute could be marked as nullable. This is annoying for the developer, but doable. It just requires extra null checks for type safety. Second, provide a `LayoutServerLoad` function wrapper to initialize the store (much like how [sveltekit-flash-message][1] does it). This option comes with its own set of drawbacks though. Namely, it requires an extra initialization, on top of the server hook. Finally, the other option is to not provide the `session` on `PageData` at all, and simply allow the developer to determine if they desire that functionality and document how to achieve it. This is probably equivalent to the nuclear option though.

Ultimately, at the end of the day, this bug is strictly relating to types. It has 0 "noticeable" effect on the application, and whether or not it will run. In fact, even with the repro repository, running `pnpm dev` throws no errors on the server or client. However, I will concede that running `pnpn check` will likely report the error because SvelteKit's type checking is quite thorough; and type errors shouldn't be overlooked because they can lead to strange, difficult to diagnose issues later in development.

**TL;DR**: The bug exists because authjs tells SvelteKit during initialization that somewhere in the server load chain, a `session` variable will be added to `PageData` (which will later become available on the `page` store imported from `"$app/stores"`). If this session variable is never defined, SvelteKit gets angry because it was told that this variable would be defined. It doesn't affect whether or not the app will compile and run, but it can cause issues later down the road.

[1]: https://github.com/ciscoheat/sveltekit-flash-message/blob/e3791e6d0d9004a98480e96037f157a2d2709355/src/lib/server.ts#L30

**Comment by eagerestwolf** [comment]
Actually, I'll just go ahead and submit the PR for doing it the optional way. That way, the SvelteKit API can remain as it is. Also, this is a super simple bug, and it'll help clear out one more SvelteKit issue. I'll also check if there are any other low hanging fruit in the issues. I'll just update this:

```ts
declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace App {
    interface Locals {
      getSession(): Promise<Session | null>
    }
    interface PageData {
      session: Session | null
    }
  }
}
```

to this:

```ts
declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace App {
    interface Locals {
      getSession(): Promise<Session | null>
    }
    interface PageData {
      session?: Session | null
    }
  }
}
```

A suggestion to make the API simpler would be to update the `getSession()` function to return `undefined` instead of `null` when no session exists, further simplifying the interface to this:

```ts
declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace App {
    interface Locals {
      getSession(): Promise<Session | undefined>
    }
    interface PageData {
      session?: Session
    }
  }
}
```

But that's probably beyond the scope of this issue.
